logger.go

```diff
@@ -46,13 +44,21 @@ func (f EmitterFunc) Emit(c context.Context, e *Entry) {
 // The default text format will have a newline appended if one is not present in
 // the message.
 type Logger struct {
-	prefix  string
-	datefmt string
-	flags   uint
-
+	caller  bool
 	emitter Emitter
 }
 
+var now = time.Now
+
+// OverrideTimestamp sets the function that will be used to get the current
+// time for each log entry.
+//
+// This is primarily meant to be used for testing custom emitters.
+// For example: OverrideTimestamp(func() time.Time { return time.Time{} })
+func OverrideTimestamp(f func() time.Time) {
```

Comment:
This should be an Option

Reply:
If it's an Option that means it won't work for testing the default Logger. Because it's meant for use with tests, I don't see a problem with making it a global setting.